lib/core/test/unit/Cardano/Wallet/DB/Properties.hs

```diff
@@ -250,7 +266,13 @@ sortedUnions = map (Identity . sort' . runIdentity) . unions
 
 -- | Execute an action once per key @k@ present in the given list
 once :: (Ord k, Monad m) => [(k,v)] -> ((k,v) -> m a) -> m [a]
-once xs = forM (Map.toList (Map.fromList xs))
+once xs action = fmap catMaybes $ flip evalStateT mempty $
+    forM xs $ \(k, v) -> do
+        s <- get
+        modify' (Set.insert k)
+        case Set.member k s of
+            False -> Just <$> lift (action (k, v))
+            True  -> pure Nothing
```

Comment:
I've changed the implementation to not use `Map` because, by doing `Map.toList` and `Map.fromList`, we were loosing some possible ordering of the keys. Now using a set instead to check if the action has already been executed.